GridTariffAPI.lib/Controllers/v1/TariffQueryController.cs

```diff
@@ -44,15 +44,23 @@ public class TariffQueryController : ControllerBase
         [Route("tariffquery")]
         [ProducesResponseType(StatusCodes.Status200OK)]
         [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
         [ProducesResponseType(typeof(ValidationProblemDetails), StatusCodes.Status400BadRequest)]
 
         public async Task<ActionResult<Models.Digin.TariffQueryResult>> TariffQuery([FromQuery] TariffQueryRequest request)  
         {
-            string validationErrorMsg = await _controllerValidationHelper.ValidateRequestInputAsync(request);
+            string validationErrorMsg =  _controllerValidationHelper.ValidateRequestInput(request);
             if (!String.IsNullOrEmpty(validationErrorMsg))
             {
                 return BadRequest(validationErrorMsg);
             }
+
+            bool test = await _controllerValidationHelper.ValidateTariffExistsAsync(request);
```

Comment:
1. rename eller droppe test-variabelen?
2. I ValidateRequestInput sjekker man at productKey og tariffKey er mutually exclusive. Men i ValidateTariffExistsAsync sjekker man begge ved oppslag. Foreslår at vi heller flytter DecideTariffKeyFromInputAsync opp mellom disse, og benytter kun tariffKey for sjekk i ValidateTariffExistsAsync.